src/picosvg/svg_reuse.py

```diff
@@ -70,3 +90,214 @@ def affine_between(
+
+    # Make path relative, with first coord at 0,0
+    x, y = _first_move(path)
+    path.move(-x, -y, inplace=True)
+
+    # Normlize vector 1 to [1 0]; eliminates rotation and uniform scaling
+    vec1 = _nth_vector(path, 1)  # ignore M 0,0
+    affine1 = _affine_vec2vec(vec1, Vector(1, 0))
+    path.walk(lambda *args: _affine_callback(affine1, *args))
+
+    # Scale first y movement to 1.0
+    vecy = _first_y(_vectors(path))
+    if vecy and not almost_equal(vecy.y, 1.0):
+        affine2 = Affine2D.identity().scale(1, 1 / vecy.y)
+        path.walk(lambda *args: _affine_callback(affine2, *args))
+
+    # TODO: what if shapes are the same but different start point
+    # TODO: what if shapes are the same but different drawing cmds
+    # This DOES happen in Noto; extent unclear
+
+    path.round_floats(tolerance, inplace=True)
+    return path
+
+
+def affine_between2(
+    s1: SVGShape, s2: SVGShape, tolerance: int = _DEFAULT_TOLERANCE
+) -> Optional[Affine2D]:
+    """Returns the Affine2D to change s1 into s2 or None if no solution was found.
+
+    Intended use is to call this only when the normalized versions of the shapes
+    are the same, in which case finding a solution is typical
+
+    """
+
+    def _apply_affine(affine, s):
+        s_prime = copy.deepcopy(s)
+        s_prime.walk(lambda *args: _affine_callback(affine, *args))
+        return s_prime
+
+    def _try_affine(affine, s1, s2):
+        return _apply_affine(affine, s1).almost_equals(s2, tolerance)
+
+    def _round(affine, s1, s2):
+        # TODO bsearch?
+        for i in _ROUND_RANGE:
+            rounded = affine.round(i)
+            if _try_affine(rounded, s1, s2):
+                return rounded
+        return affine  # give up
+
+    s1 = dataclasses.replace(s1, id="")
+    s2 = dataclasses.replace(s2, id="")
+
+    if s1.almost_equals(s2, tolerance):
+        return Affine2D.identity()
+
+    s1 = _affine_friendly(s1)
+    s2 = _affine_friendly(s2)
+
+    s1x, s1y = _first_move(s1)
+    s2x, s2y = _first_move(s2)
+
+    affine = Affine2D.identity().translate(s2x - s1x, s2y - s1y)
+    if _try_affine(affine, s1, s2):
+        return affine
+
+    # Normalize first edge.
+    # Fixes rotation, x-scale, and uniform scaling.
+    s1_vec1 = _nth_vector(s1, 1)
+    s2_vec1 = _nth_vector(s2, 1)
+
+    s1_to_origin = Affine2D.identity().translate(-s1x, -s1y)
+    s2_to_origin = Affine2D.identity().translate(-s2x, -s2y)
+    s1_vec1_to_s2_vec2 = _affine_vec2vec(s1_vec1, s2_vec1)
+
+    # Move to s2 start
+    origin_to_s2 = Affine2D.identity().translate(s2x, s2y)
+
+    affine = Affine2D.compose_ltr((s1_to_origin, s1_vec1_to_s2_vec2, origin_to_s2))
+    if _try_affine(affine, s1, s2):
+        return _round(affine, s1, s2)
+
+    # Could be non-uniform scaling or mirroring
+    # Try to match up the first y movement
+
+    # Could be non-uniform scaling and/or mirroring
+    # Scale first y movement (after matching up vec1) to match
```

Comment:
the two comments above look like they say similar things? one seems redundant

Reply:
I don't often comment but when I do I do it twice ... fixed